src/devices/Pn5180/Pn5180.cs

```diff
@@ -0,0 +1,1496 @@
+            // The parameter 0x08 to have 1 slot
+            // then CRC, normal value for this buffer are already in it
+            Span<byte> wupb = stackalloc byte[3]
+            {
+                0x05,
+                0x00,
+                0x08
+            };
+            // Will get the ATQB answer
+            Span<byte> atqb = stackalloc byte[12];
+            // For this part, activate CRC
+            CrcReceptionTransfer = true;
+            int numBytes;
+
+            DateTime dtTimeout = DateTime.Now.AddMilliseconds(timeoutPollingMilliseconds);
+
+            try
+            {
+                do
+                {
+                    // Clears all interrupt
+                    SpiWriteRegister(Command.WRITE_REGISTER, Register.IRQ_CLEAR, new byte[] { 0xFF, 0xFF, 0x0F, 0x00 });
+                    // Sets the PN5180 into IDLE state
+                    SpiWriteRegister(Command.WRITE_REGISTER_AND_MASK, Register.SYSTEM_CONFIG, new byte[] { 0xF8, 0xFF, 0xFF, 0xFF });
+                    // Activates TRANSCEIVE routine
+                    SpiWriteRegister(Command.WRITE_REGISTER_OR_MASK, Register.SYSTEM_CONFIG, new byte[] { 0x03, 0x00, 0x00, 0x00 });
+                    // Sends REQB command
+                    ret &= SendDataToCard(wupb);
+                    (numBytes, _) = GetNumberOfBytesReceivedAndValidBits();
+                    if (numBytes > 0)
+                    {
+                        ret &= ReadDataFromCard(atqb, atqb.Length);
+                        if (!ret)
+                        {
+                            return false;
+                        }
+                    }
+                    else if (dtTimeout < DateTime.Now)
+                    {
+                        return false;
+                    }
+                }
+                while (numBytes == 0);
+
+                card = new Data106kbpsTypeB(atqb.ToArray());
+
+                card.TargetNumber = targetNumber;
+                // Max target is 14, can't exceed this
+                if (card.TargetNumber > 14)
+                {
+                    // We still output the data but we let
+                    // the application know that the card can't be selected
+                    return false;
+                }
+
+                // Now all communications will embedd the CRC
+                CrcReceptionTransfer = true;
+                ret = SelectCardTypeB(card);
+                if (!ret)
+                {
+                    // If we can't select the card, we still output the data but we let
+                    // the application know that the card can't be selected
+                    return false;
+                }
+
+                // Add the card to the list
+                var selected = new SelectedPiccInformation() { Card = card, LastBlockMark = false };
+                _activeSelected.Add(selected);
```

Comment:
do we ever remove it from the list?

Reply:
yes, in the test right before if the card is not seen or not responding.